app/post.go

```diff
@@ -273,8 +277,9 @@ func (a *App) CreatePost(c request.CTX, post *model.Post, channel *model.Channel
 				return false
 			}
 			if replacementPost != nil {
-				// the original post's metadata (if there ever was any) is lost, and will be rebuilt.
 				post = replacementPost
+				// we nee the metadata to persist, we are saving metadata.priority in posts priority table
+				post.Metadata = metadata
```

Comment:
@lieut-data we need to reinstate the metadata, since that's the field we are using to save post priority values.
Do you see any reason that this could cause problems?
Do plugins set their own `metadata`, would we need to merge `PostMetadata.Priority`?
cc @lieut-data 

BTW, `post.ForPlugin` clones the post, and then nils the post.Metadata for the clone.
Since cloning is a shallow copy, wouldn't that make the original post's metadata nil?
I wonder if we mind to deep copy metadata in the `post.Clone` method

Reply:
Hey @koox00! There's potentially a couple of layers to this question, but it all sounds like we need to rethink what it means to allow a plugin to replace post as part of processing the `MessageWillBePosted`. Are some fields off limits, and the server will forcibly rebuild them? Or can plugins act unilaterally in this regard?

cc @catalintomai / @levb

Reply:
The most outward "layer" is that `Metadata` was a rather extensible struct, containing 3rd party types. When these types change, we end up causing backwards compatibility issues with previously compiled plugins, leading to crashes.

One solution would be to separate the "plugin-visible" metadata into another struct, and maintain the type version discipline on the new struct. We wold **still** have issues with older plugins, so this would be a "new API" change?

FWIW, I am not sure that restoring the old posts' metadata to the modified post is the right thing to do.

Reply:
> FWIW, I am not sure that restoring the old posts' metadata to the modified post is the right thing to do.

@levb would restoring the `metadata.priority` make sense? Since that's what we want to handle through the app and not a plugin.